build_tools/lint.fish

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env fish
+#
+# This is meant to be run by "make lint" or "make lint-all". It is not meant to
+# be run directly from a shell prompt.
+#
+set cppchecks warning,performance,portability,information,missingInclude
+set cppcheck_args
+set c_files
+set all no
+
+set -gx CXX $argv[1]
+set -e argv[1]
+
+if test "$argv[1]" = "--all"
+    set all yes
+    set c_files src/
+    set cppchecks "$cppchecks,unusedFunction"
+    set -e argv[1]
+end
+
+# We only want -D and -I options to be passed thru to cppcheck.
+for arg in $argv
+    if string match -q -- '-D*' $arg
+        set cppcheck_args $cppcheck_args $arg
+    else if string match -q -- '-I*' $arg
+        set cppcheck_args $cppcheck_args $arg
+    end
+end
+if test (uname -m) = "x86_64"
+    set cppcheck_args -D__x86_64__ -D__LP64__ $cppcheck_args
+end
+
+if test $all = no
+    # We haven't been asked to lint all the source. If there are uncommitted
+    # changes lint those, else lint the files in the most recent commit.
+    set pending (git status --short --untracked-files=all | sed -e 's/^ *//')
```

Comment:
What's the sed for? With my git, it seems to do nothing. If the intention is to strip whitespace, you could also do `string trim`. Alternatively, it's probably more stable to use `git status --porcelain` (which is guaranteed stable and ignores a few configuration options) and strip everything up to and including the first space.